# 2019年5月21日（火曜日）
# プログラミング基礎I課題

### 注意事項
* 次回以降、プログラミング基礎1の講義や演習で新しくノートブックを作成したりアップロードしたりするときは、 
左のサイドバーで "prokiso1-2019" のフォルダに移動してから行おう。
* 左のサイドバーで上位フォルダに移動するには、上部にあるフォルダ名をクリックすれば移動できる。

* **その他にも、様々な機能が JupyterLab にはある。その操作を全て講義で説明することはできないが、知っていると便利な機能も沢山あるので、家で自分で調べておこう。**



## 課題1(大文字小文字の変換)

### 課題1-1

以下の文を文字列s1に代入し、文字列s1の文字をすべて小文字にせよ。

>IoT と Blockchain

【実行結果】
>iot と blockchain

In [1]:
s1 = "IoT と Blockchain"
print(s1.lower())

iot と blockchain


### 課題1-2

文字列s1の大文字と小文字を入れ替えよ。

【実行結果】
>iOt と bLOCKCHAIN

In [2]:
print(s1.swapcase())

iOt と bLOCKCHAIN


### 課題1-3

以下の文を文字列s2に代入し、文字列s2の各単語の1文字目を大文字にし、単語の2文字目以降を小文字にせよ。

what is not fully understood is not possessed.

【実行結果】
>What Is Not Fully Understood Is Not Possessed.

In [3]:
s2 = "what is not fully understood is not possessed."
print(s2.title())

What Is Not Fully Understood Is Not Possessed.


## 課題2(文字列の検索、置換、除去)
## 課題2-1
以下の文を文字列s3に代入する。

192-0982, Katakuramachi, Hachioji City, Tokyo/////

In [4]:
s3 = "192-0982, Katakuramachi, Hachioji City, Tokyo/////"

### 課題2-2
文字列s3に文字「a」が何個含まれてるかを出力せよ。

In [5]:
print(s3.count("a"))

5


### 課題2-3
文字列s3から文字「Tokyo」が最初に出てくるindex値を出力せよ。

In [6]:
print(s3.find("Tokyo"))

40


### 課題2-4
文字列s3から「Tokyo」を「Japan」に置換して出力せよ。

In [7]:
print(s3.replace("Tokyo", "Japan"))

192-0982, Katakuramachi, Hachioji City, Japan/////


### 課題2-5
文字列s3から末尾の連続するスラッシュを取り除いて出力せよ。

In [8]:
print(s3.rstrip("/"))

192-0982, Katakuramachi, Hachioji City, Tokyo


### 課題2-6
文字列s3からカンマを取り除いて出力せよ。

In [9]:
print(s3.replace(",",""))

192-0982 Katakuramachi Hachioji City Tokyo/////


### 課題2-7
文字列s3から192を抜き出し、int型にして500を加算して出力せよ。

In [10]:
print(int(s3[:3]) + 500)

692


## 課題3(文字列に値の埋め込み)
### 課題3-1
自分の学籍番号をid、氏名をname、年齢をageにそれぞれ代入せよ。


In [2]:
id = 999
name = "koka taro"
age = 18

### 課題3-2
以下のような実行結果となるよう、f"{値}"の書式を使う方法で表示せよ。

【実行結果】
>学籍番号：999<br>
>氏名　　：koka taro<br>
>年齢　　：18

In [4]:
text1 = f"学籍番号：{id}\n氏名　　：{name}\n年齢　　：{age}"
print(text1)

学籍番号：999
氏名　　：koka taro
年齢　　：18


### 課題3-3
以下のような実行結果となるよう、format()を使う方法で表示せよ。

【実行結果】
>氏名：koka taro、学籍番号：999、年齢：18

In [19]:
s4 = "氏名：{}、学籍番号：{}、年齢：{}"
text2 = s4.format(name, id, age)
print(text2)

氏名：koka taro、学籍番号：999、年齢：18


## 課題4(値の書式設定)
### 課題4-1
埼玉県の人口は7,322,645人、面積は3,797.75$km^2$であり、<br>
千葉県の人口は6,268,585人、面積は5,157.61$km^2$であり、<br>
愛知県の人口は7,539,185人、面積は5,172.96$km^2$である。<br>
埼玉県、千葉県、愛知県、それぞれの人口密度[人/$km^2$]を求め、以下のような実行結果となるプログラムを作成せよ。

【実行結果】
>埼玉の人口密度：1928.1535119478638 <br>
千葉の人口密度：1215.4050034802942 <br>
愛知の人口密度：1457.4218629179425

In [11]:
saitama_population = 7322645
saitama_area = 3797.75
chiba_population = 6268585 
chiba_area = 5157.61
aichi_population = 7539185
aichi_area = 5172.96
saitama_population_density = saitama_population / saitama_area
chiba_population_density = chiba_population / chiba_area
aichi_population_density = aichi_population / aichi_area
print("埼玉の人口密度：" + str(saitama_population_density))
print("千葉の人口密度：" + str(chiba_population_density))
print("愛知の人口密度：" + str(aichi_population_density))

埼玉の人口密度：1928.1535119478638
千葉の人口密度：1215.4050034802942
愛知の人口密度：1457.4218629179425


### 課題4-2
以下のような実行結果となるよう、f"{値}"の書式を使う方法で表示せよ。小数点以下2位までを表示する。

【実行結果】
>埼玉県：1,928.15、愛知県：1,457.42、千葉県：1,215.41


In [23]:
text = f"埼玉県：{saitama_population_density:,.2f}、愛知県：{aichi_population_density:,.2f}、千葉県：{chiba_population_density:,.2f}"
print(text)

埼玉県：1,928.15、愛知県：1,457.42、千葉県：1,215.41


### 課題4-3
以下のような実行結果となるよう、format()を使う方法で表示せよ。小数点以下2位までを表示する。

【実行結果】
>埼玉県：1,928.15、愛知県：1,457.42、千葉県：1,215.41

In [13]:
s5 = "埼玉県：{:,.2f}、愛知県：{:,.2f}、千葉県：{:,.2f}"
print(s5.format(saitama_population_density, aichi_population_density,chiba_population_density))

埼玉県：1,928.15、愛知県：1,457.42、千葉県：1,215.41
